docs/temporal.md

```diff
@@ -0,0 +1,229 @@
+2. As explained above, the workflow represents a deterministic piece of code that can use non-deterministic activities for operations that require IO.
+3. [`TemporalAgent.run()`][pydantic_ai.durable_exec.temporal.TemporalAgent.run] works just like [`Agent.run()`][pydantic_ai.Agent.run], but it will automatically offload model requests, tool calls, and MCP server communication to Temporal activities.
+4. We connect to the Temporal server which keeps track of workflow and activity execution.
+5. This assumes the Temporal server is [running locally](https://github.com/temporalio/temporal#download-and-start-temporal-server-locally).
+6. The [`PydanticAIPlugin`][pydantic_ai.durable_exec.temporal.PydanticAIPlugin] tells Temporal to use Pydantic for serialization and deserialization, and to treat [`UserError`][pydantic_ai.exceptions.UserError] exceptions as non-retryable.
+7. We start the worker process that will listen on the specified task queue and run workflows and activities.
+8. The [`AgentPlugin`][pydantic_ai.durable_exec.temporal.AgentPlugin] registers the `TemporalAgent`'s activities with the worker.
+9. We call on the server to execute the workflow on a worker that's listening on the specified task queue.
+10. The agent's `name` is used to uniquely identify its activities.
+
+_(This example is complete, it can be run "as is" — you'll need to add `asyncio.run(main())` to run `main`)_
+
+In a real world application, the agent, workflow, and worker are typically defined separately from the code that calls for a workflow to be executed.
+Because Temporal workflows need to be defined at the top level of the file and the `TemporalAgent` instance is needed inside the workflow and when starting the worker (to register the activities), it needs to be defined at the top level of the file as well.
+
+For more information on how to use Temporal in Python applications, see their [Python SDK guide](https://docs.temporal.io/develop/python).
+
+## Agent and Toolset Requirements
+
+To ensure that Temporal knows what code to run when an activity fails or is interrupted and then restarted, even if your code is changed in between, each activity needs to have a name that's stable and unique.
+
+When `TemporalAgent` dynamically creates activities for the wrapped agent's model requests and toolsets (specifically those that implement their own tool listing and calling, i.e. [`FunctionToolset`][pydantic_ai.toolsets.FunctionToolset] and [`MCPServer`][pydantic_ai.mcp.MCPServer]), their names are derived from the agent's [`name`][pydantic_ai.agent.AbstractAgent.name] and the toolsets' [`id`s][pydantic_ai.toolsets.AbstractToolset.id]. These fields are normally optional, but are required to be set when using Temporal. They should not be changed once the temporal agent has been deployed to production as this would break active workflows.
+
+Other than that, any agent and toolset will just work!
+
+## Agent Run Dependencies
```

Comment:
I think we should really try to reduce this boilerplate for users if at all possible. We can come back to it after merging if necessary, but it feels to me like it may be possible to do this more automatically and/or at least make it error at activity-creation time rather than at activity-run time (when you access non-serialized fields).